py/modmath.c

```diff
@@ -196,7 +196,17 @@ MATH_FUN_1(erf, erf)
 // erfc(x): return the complementary error function of x
 MATH_FUN_1(erfc, erfc)
 // gamma(x): return the gamma function of x
+#if MICROPY_PY_MATH_GAMMA_FIX_NEG_INF
+static mp_float_t tgamma_funcf(mp_float_t x) {
+    if (__builtin_isinf_sign(x) == -1) {
```

Comment:
We don't use builtin's for floats anywhere else.  Is this portable?  Can we use `isinf(x) && x < 0` instead?  It may compile to the same thing anyway.

Reply:
Yeah, I settled on this because in theory `isinf(x) == -1` should work but it doesn't on ESP32. `isinf` is actually a BSD extension, the actual C99 version is way more verbose to test the same thing: https://www.gnu.org/software/libc/manual/html_node/Floating-Point-Classes.html

However gcc does ship a builtin for BSD `isinf`, so I used that.

But your suggestion seems good, will change it.